### PR TITLE
[wasm] Generate the pinvoke tables.

### PR DESCRIPTION
**Pull request by vargaz**
Put the default non-linked version of the table into version control, usable by the interpreter only build.

### CODE REVIEW

File: sdks/wasm/pinvoke-tables-default.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
+typedef struct {
```
Comment (jaykrell) [inline sdks/wasm/pinvoke-tables-default.h]: nit i.e. ignorable: Please add a comment saying this is generated, if it is, as I suspect. The generator should output the comment.

File: mcs/tools/wasm-tuner/tuner.cs [hunk 2]
```diff
@@ -76,12 +89,94 @@ public class WasmTuner
 		}
 	}
 
+	void Usage () {
+		Console.WriteLine ("Usage: tuner.exe <arguments>");
+		Console.WriteLine ("Arguments:");
+		Console.WriteLine ("--gen-icall-table icall-table.json <assemblies>.");
+		Console.WriteLine ("--gen-pinvoke-table <list of native library names separated by commas> <assemblies>.");
+	}
```
Comment (kjpou1) [inline mcs/tools/wasm-tuner/tuner.cs]: The `Usage:` seems to suggest that one can specify both `--gen-icall-table` and `--gen-pinvoke-table` arguments but below it it seems to to only process `--gen-icall-table` or `--gen-pinvoke-table`.  Is that correct?

Comment (vargaz) [reply mcs/tools/wasm-tuner/tuner.cs]: Currently yes, later we want both of them so it only needs to run once.

**Comment by vargaz** [comment]
@monojenkins build failed

File: sdks/wasm/pinvoke-tables-default.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+{"SystemNative_Rename", SystemNative_Rename},
+{"SystemNative_RmDir", SystemNative_RmDir},
+{"SystemNative_Stat2", SystemNative_Stat2},
+{"SystemNative_LStat2", SystemNative_LStat2},
+{"SystemNative_UTime", SystemNative_UTime},
+{"SystemNative_UTimes", SystemNative_UTimes},
+{"SystemNative_Unlink", SystemNative_Unlink},
+{NULL, NULL}
+};
+static void *pinvoke_tables[] = { System_Native_imports,};
```
Comment (jaykrell) [inline sdks/wasm/pinvoke-tables-default.h]: nit (again, ignorable), you already know: a type other than `void*` would be nice

File: sdks/wasm/packager.cs [hunk 4]
```diff
@@ -721,6 +721,10 @@ class WasmOptions {
 			ninja.WriteLine ($"build $builddir/driver.c: cpifdiff {source_file}");
 			ninja.WriteLine ($"build $builddir/driver-gen.c: cpifdiff $builddir/driver-gen.c.in");
 
+			var pinvoke_file = Path.GetFullPath (Path.Combine (tool_prefix, "pinvoke-tables-default.h"));
+			ninja.WriteLine ($"build $builddir/pinvoke-tables-default.h: cpifdiff {pinvoke_file}");
+			driver_deps += $" $builddir/pinvoke-tables-default.h";
```
Comment (jaykrell) [inline sdks/wasm/packager.cs]: nit, ignorable: first dollar sign is presumably nop

File: mcs/tools/wasm-tuner/tuner.cs [hunk 5]
```diff
@@ -76,12 +89,94 @@ public class WasmTuner
...
+			string symbol = module.Replace (".", "_") + "_imports";
+			Console.WriteLine ("static PinvokeImport " + symbol + " [] = {");
+			foreach (var pinvoke in pinvokes) {
+				if (pinvoke.Module == module)
+					Console.WriteLine ("{\"" + pinvoke.EntryPoint + "\", " + pinvoke.EntryPoint + "},");
+			}
+			Console.WriteLine ("{NULL, NULL}");
+			Console.WriteLine ("};");
+		}
+		Console.Write ("static void *pinvoke_tables[] = { ");
```
Comment (jaykrell) [inline mcs/tools/wasm-tuner/tuner.cs]: Nit, ignorable, you know, non-`void*` preferable, and `const`. `const` provides efficiency in most C/C++ but maybe not in wasm.

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**